Cubical/Foundations/Prelude.agda

```diff
@@ -61,35 +61,112 @@ cong₂ : ∀ {C : (a : A) → (b : B a) → Type ℓ} →
         PathP (λ i → C (p i) (q i)) (f x u) (f y v)
 cong₂ f p q i = f (p i) (q i)
 
--- The filler of homogeneous path composition:
--- compPath-filler p q = PathP (λ i → x ≡ q i) p (p ∙ q)
+{- The most natural notion of homogenous path composition
+    in a cubical setting is double composition:
 
-compPath-filler : ∀ {x y z : A} → x ≡ y → y ≡ z → I → I → A
-compPath-filler {x = x} p q j i =
-  hfill (λ j → λ { (i = i0) → x
-                  ; (i = i1) → q j }) (inS (p i)) j
+     x ∙ ∙ ∙ > w
```

Comment:
It's a bit weird to have p go in this direction... Why not have p^⁻1 and have it look like a normal composition diagram?

Reply:
I actually find it easier to visualize squares/cubes/etc. this way (drawing p^-1 as an arrow going the opposite direction) but happy to change it if it's nonstandard or less intuitive to others.

Reply:
I guess it's OK in 2D, but it doesn't scale very well to higher dimensions... At least I get very confused when the arrows all go in different direction when you draw a cube. Plus, this drawing doesn't really match the semantic meaning of hcomp as an operation that gives you the missing side of an open box.

Reply:
Ah, these are very good points. I should probably stop thinking about the edges of cubes this way.